react/features/base/participants/actions.ts

```diff
@@ -253,6 +254,39 @@ export function participantJoined(participant: IParticipant) {
     };
 }
 
+/**
+ * Updates the sources of a remote participant.
+ *
+ * @param {IJitsiParticipant} jitsiParticipant - The IJitsiParticipant instance.
+ * @returns {{
+ *      type: PARTICIPANT_SOURCES_UPDATED,
+ *      participant: IParticipant
+ * }}
+ */
+export function participantSourcesUpdated(jitsiParticipant: IJitsiParticipant) {
+    return (dispatch: IStore['dispatch'], getState: IStore['getState']) => {
+        const id = jitsiParticipant.getId();
+        const participant = getParticipantById(getState(), id);
+
+        if (participant?.local) {
+            return;
+        }
+        const sources = jitsiParticipant.getSources();
+
+        if (!sources?.size) {
```

Comment:
Why do we have this check? What will happen for the use case where we don't receive media for a participant anymore? How will we remove its sources?

Reply:
Currently, LJM doesn't support removal of sources. Once a source is signaled, they can only be muted. Even when we remove the tracks from the peerconnection, the SSRC and the source-name will be the same since we do not want extra signaling to be exchanged back and forth when a different track gets added back to the peerconnection. When we stop receiving media, the muted state will be updated for the source.